wagtail/wagtailadmin/static_src/wagtailadmin/js/modal-workflow.js

```diff
@@ -51,9 +51,15 @@ function ModalWorkflow(opts) {
     };
 
     self.loadResponseText = function(responseText) {
-        var response = eval('(' + responseText + ')');
+        var response;
 
-        self.loadBody(response);
+        try {
+            response = JSON.parse( responseText );
+        } catch ( error ) {
+            window.location = window.location.host;
```

Comment:
ok, I'm going to go  the `"your session has timed out, please log in in another tab" ` route. Thanks for the feedback. I will also work on removing `eval` and `new Function`, as both are insecure and also pollute the global namespace.